prow/apis/prowjobs/v1/types.go

```diff
@@ -697,6 +707,17 @@ func (d *DecorationConfig) ApplyDefault(def *DecorationConfig) *DecorationConfig
 		merged.DefaultMemoryRequest = def.DefaultMemoryRequest
 	}
 
+	if merged.PodPendingTimeout == nil {
+		merged.PodPendingTimeout = def.PodPendingTimeout
+	}
+
+	if merged.PodRunningTimeout == nil {
+		merged.PodRunningTimeout = def.PodRunningTimeout
+	}
+
+	if merged.PodUnscheduledTimeout == nil {
+		merged.PodUnscheduledTimeout = def.PodUnscheduledTimeout
+	}
```

Comment:
Maybe I'm missing something, but how are the values consumed? What actually enforces these timeouts?

Reply:
> Maybe I'm missing something, but how are the values consumed? What actually enforces these timeouts?

Prow allows setting DecorationConfig on both prowjob yaml and prow config yaml, and the values of these two are merged at runtime by ApplyDefault function https://github.com/kubernetes/test-infra/blob/959d4500863c43efa5cb45b49ba6d14c62441103/prow/apis/prowjobs/v1/types.go#L615 
@chaodaiG please correct me if I'm wrong